scripts/train.py

```diff
@@ -25,7 +25,7 @@ def main():
     except KeyboardInterrupt:
         canceled = True
 
-    if not canceled or args.backup_before_save:
+    if not canceled:
```

Comment:
I'm not sure about this change. I originally checked for `or args.backup_before_save` because I still wanted to get a backup even if I cancel the training script. But I think it will also save the model when canceling now. Maybe something like this would be better:

```python
    if not canceled:
        trainer.end()
    elif args.backup_before_save:
        trainer.backup()
```

or you could just call `trainer.end()` without any check. I'm honestly not sure what the best solution is here.

Reply:
Ah, that was quite the miss on my part, I'm going to revert it to what it was before since this isn't related at all to the not saving bug I was trying to fix.

Reply:
I wouldn't worry too much about saving the model on cancelling, my own preference would be to have the model made, if it's not wanted it's easy to delete.